Protobuild.Manager/LinuxProgram.cs

```diff
@@ -41,111 +49,52 @@ public static void Main(string[] args)
+                catch (Exception e)
+                { 
+                    // we don't want an infinite loop
+                    if (args.ToList().Contains("--internalbreak"))
+                        return;
+                    
+                    if (e is FileNotFoundException || e is TypeInitializationException)
                     {
-                        // Try and extract webkit-sharp next to the assembly.
-                        try
+                        foreach (var ed in embeded_dlls)
                         {
-                            var assembly = typeof(Program).Assembly.GetManifestResourceStream("webkit-sharp.dll");
-                            var assemblyConfig = typeof(Program).Assembly.GetManifestResourceStream("webkit-sharp.dll.config");
-
-                            using (var writer = new FileStream(Path.Combine(new FileInfo(typeof(Program).Assembly.Location).DirectoryName, "webkit-sharp.dll"), FileMode.Create, FileAccess.Write))
-                            {
-                                assembly.CopyTo(writer);
-                            }
-                            using (var writer = new FileStream(Path.Combine(new FileInfo(typeof(Program).Assembly.Location).DirectoryName, "webkit-sharp.dll.config"), FileMode.Create, FileAccess.Write))
-                            {
-                                assemblyConfig.CopyTo(writer);
-                            }
-
-                            // We have to relaunch the executable to pick up the new libraries next to us.
-                            var process = System.Diagnostics.Process.Start("mono", "\"" + typeof(Program).Assembly.Location + "\" " + string.Join(" ", args));
-                            process.WaitForExit();
-
-                            if (process.ExitCode == 1)
-                            {
-                                File.Delete(Path.Combine(new FileInfo(typeof(Program).Assembly.Location).DirectoryName, "webkit-sharp.dll"));
-                                File.Delete(Path.Combine(new FileInfo(typeof(Program).Assembly.Location).DirectoryName, "webkit-sharp.dll.config"));
-                                ShowRequiredLibrary(new[] {
-                                    "zypper in webkit-sharp"
-                                });
-                            }
-
-                            Environment.Exit(process.ExitCode);
-                        }
-                        catch
-                        {
-                            ShowRequiredLibrary(new[] {
-                                "zypper in webkit-sharp"
-                            });
+                            WriteResource(ed, true);
+                            WriteResource(ed + ".config", true);
+
+                            // TODO Include native .so libraries and extract the ones 
+                            // needed for the current PC architecture
                         }
+
+                        if (!ReRun(args))
```

Comment:
You need to do `Environment.Exit(process.ExitCode)` so that the exit code of the inner-most process gets passed up, rather than writing out the error to console.